Bookish/Controllers/BookController.cs

```diff
@@ -152,8 +177,45 @@ public async Task<ActionResult> BookInput(BookInput input)
         }
         return View();
     }
+
+    [HttpGet]
+    public async Task<ActionResult> BookEdited(Book input)
+    {
+        string bookId = input.Id;
+        using (var context = new EFCore())
+        {
+            var book = context.Books.Find(bookId);
+            if (book != null)
+            {
+                if (input.Title != null) book.Title = input.Title;
+                if (input.Author != null) book.Author = input.Author;
+                context.SaveChanges();
+                return View();
+            }
+
+            return View("BookNotEdited");
+        }
+    }
     
-    
+    [HttpGet]
+    public async Task<ActionResult> BookDeleted(Book input)
+    {
+        string bookId = input.Id;
+        using (var context = new EFCore())
+        {
+            var book = context.Books.Find(bookId);
```

Comment:
Same again - I tend to write my methods in a pattern of:
- get input
- if some conditions aren't satisfied, early return with an error view/redirect to a different page etc
- otherwise (if everything's okay), do the thing and return the view